### PR TITLE
Update axon version - plus cleaning

### PR DESCRIPTION
**Pull request by meletios**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -150,12 +150,12 @@
     <dependency>
       <groupId>org.axonframework</groupId>
       <artifactId>axon-spring-boot-starter</artifactId>
-      <version>3.3.7</version>
+      <version>4.2</version>
```
Comment (palmenhq) [inline pom.xml]: It's so strange that they don't have the patch number in this version but it seems like they _actually_ published `4.2`

File: src/main/java/com/hedvig/productPricing/service/configuration/AxonConfiguration.java [hunk 2]
```diff
@@ -102,11 +120,10 @@ Executor executor() {
     return Executors.newFixedThreadPool(3);
   }
 
-  @Bean
-  SagaConfiguration<ContractSignedSaga> contractSignedSagaConfiguration(Executor executor) {
-    return SagaConfiguration.subscribingSagaManager(
+  @Autowired
```
Comment (palmenhq) [inline src/main/java/com/hedvig/productPricing/service/configuration/AxonConfiguration.java]: does this work though? Just curious

File: src/test/java/com/hedvig/productPricing/AxonTestConfiguration.java [hunk 3]
```diff
@@ -10,39 +10,25 @@
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.context.annotation.Bean;
 
-//@ComponentScan("com.hedvig.productPricing")
```
Comment (palmenhq) [inline src/test/java/com/hedvig/productPricing/AxonTestConfiguration.java]: nice

File: src/test/java/com/hedvig/productPricing/AxonTestConfiguration.java [hunk 4]
```diff
@@ -10,39 +10,25 @@
...
-
-    @Bean
-    public EventStorageEngine axonConfiguration() {
-
-        InMemoryEventStorageEngine inMemoryEventStorageEngine = new InMemoryEventStorageEngine();
-        return inMemoryEventStorageEngine;
-    }
-
-
-    /*
```
Comment (palmenhq) [inline src/test/java/com/hedvig/productPricing/AxonTestConfiguration.java]: 😍

File: src/test/java/com/hedvig/productPricing/AxonTestConfiguration.java (hunk 3 above)
Comment (meletios) [reply src/test/java/com/hedvig/productPricing/AxonTestConfiguration.java]: 😎

**Pull request merged**